sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/OpenCoverReportParserTest.java

```diff
@@ -129,16 +130,56 @@ public void branchCoverage() throws Exception {
         Assertions.entry(18, 1),
         Assertions.entry(25, 1));
 
-    assertThat(coverage.getBranchCoverage(filePath).size()).isEqualTo(5);
+    assertThat(coverage.getBranchCoverage(filePath))
+      .hasSize(5)
+      .contains(
+        new BranchCoverage(9, 4 , 2),
```

Comment:
for the record, the test file is [here](https://github.com/SonarSource/sonar-dotnet/blob/aa05edb2b401dacac9cc1423422fcc8729623f13/sonar-dotnet-shared-library/src/test/resources/opencover/coverage_branches.xml)

this puzzles me, line 9 is:
```
  if (x == 0 || y < 0)
```

From the ccov report:
```
<BranchPoint vc="0" uspid="11" ordinal="0" offset="2" sl="9" path="0" offsetend="4" fileid="1" />
<BranchPoint vc="1" uspid="12" ordinal="1" offset="2" sl="9" path="1" offsetend="10" fileid="1" />
<BranchPoint vc="1" uspid="13" ordinal="2" offset="13" sl="9" path="0" offsetend="15" fileid="1" />
<BranchPoint vc="0" uspid="14" ordinal="3" offset="13" sl="9" path="1" offsetend="22" fileid="1" />
```

The 4 conditions, I imagine:
- offset 2-4 - the IF - why not executed? because not all branches are executed?
- offset 2-10 - the IF and `x==0` - executed as first instruction in the OR
- offset 13-15 - the OR itself - executed with short-circuit
- offset 13-22 - the OR and `y < 10` - not executed because of short-circuit

I think we parse it well, however it's slightly confusing.

Reply:
(it wouldn't hurt to add this as a comment - we will rarely change these tests and when we will, it will ease understanding them)

Reply:
I've added the code as comment before each BranchCoverage entry in test. I know that OpenCover computes the conditions by looking at the generated IL but I'm not sure how and I don't want to make the wrong assumptions.

Reply:
I think that the cases are x== 0, x != 0, y <0  and y>0 and it matches the number of branch points...